server/src/main/java/org/elasticsearch/search/vectors/ESKnnFloatVectorQuery.java

```diff
@@ -31,8 +48,43 @@ protected TopDocs mergeLeafResults(TopDocs[] perLeafResults) {
         return topK;
     }
 
+    @Override
+    protected TopDocs approximateSearch(
+        LeafReaderContext context,
+        Bits acceptDocs,
+        int visitedLimit,
+        KnnCollectorManager knnCollectorManager
+    ) throws IOException {
+        TopDocs topDocs = super.approximateSearch(context, acceptDocs, visitedLimit, knnCollectorManager);
+        if (rescoreOversample == null) {
+            return topDocs;
+        }
+
+        BitSet exactSearchAcceptDocs = topDocsToBitSet(topDocs, context.reader().maxDoc());
+        BitSetIterator bitSetIterator = new BitSetIterator(exactSearchAcceptDocs, topDocs.scoreDocs.length);
+        QueryTimeout queryTimeout = null;
+        if (knnCollectorManager instanceof TimeLimitingKnnCollectorManager timeLimitingKnnCollectorManager) {
+            queryTimeout = timeLimitingKnnCollectorManager.getQueryTimeout();
+        }
+        return exactSearch(context, bitSetIterator, queryTimeout);
```

Comment:
I don't think this will work as `exactSearch` would use the quantized scorers, also this would be done per segment. Which would be bad.

I think what we need to do is override `rewrite` to return another query that can be further rewritten but scores the previously scored documents given the raw floating point vectors.

We only want to rescore for the entire shard. Doing each segment would be very expensive.

Reply:
I see, thanks for explaining Ben. I'll give it another shot 🔫